railties/lib/rails/generators/rails/app/templates/bin/setup.tt

```diff
@@ -27,8 +28,13 @@ chdir APP_ROOT do
   #   cp 'config/database.yml.sample', 'config/database.yml'
   # end
 
-  puts "\n== Preparing database =="
-  system! 'bin/rails db:setup'
+  if system "bin/rails db:version > /dev/null &>2"
```

Comment:
Any suggestion of a better way to test whether the database already exists would be welcome! 🙏

Reply:
Maybe there's a rake task / rails command to be extracted here, that handles this? `rails db:yes_please_do_the_thing` and let _it_ figure out whether to `setup` or `migrate`.

Reply:
`rails db:migrate_or_setup`? 😅

Reply:
Hah, I went for an obviously-ridiculous one because that's the best I could come up with too. :grin:

One other thought was `db:update`, but replacing a `bin/update` with a `bin/setup` that calls `db:update` instead of `db:setup` might feel a bit too silly.

If we can get some consensus that it's a useful thing to have, though, finding something to call it should be surmountable.

Reply:
wdyt about `db:exists` returning exit code 1 if the DB doesn't exist?

Right now I'm using it in a setup script:

```bash
#!/bin/sh

# Exit on fail
set -e

bundle check || bundle install
yarn check || yarn
rails db:exists && rails db:migrate || rails db:setup
```

Reply:
I'm not sure how we'd make a `db:exists` work with multiple databases in the future, and I'm wary that each `rails` command is pretty slow to run (e.g., we have to boot the full app to read `database.yml`, because it may contain ERb tags).

I don't particularly object to having such a command, but if the 99% use case is `db:exists ? db:migrate : db:setup`, then ISTM we should just bundle that into a single unit (which could then consider each database in turn in a multi-DB configuration).

Reply:
😍